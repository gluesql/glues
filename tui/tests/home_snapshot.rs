use color_eyre::Result;
use insta::assert_debug_snapshot;
mod common;
use common::{DEFAULT_HEIGHT, DEFAULT_QUIET_MS, DEFAULT_TIMEOUT_MS, DEFAULT_WIDTH, TuiHarness};

#[test]
fn home_screen_snapshot() -> Result<()> {
    let mut h = TuiHarness::spawn(DEFAULT_WIDTH, DEFAULT_HEIGHT)?;
    h.wait_for("Show keymap")?;
    h.wait_for("\\[q\\] Quit")?;
    h.drain_until_quiet(
        std::time::Duration::from_millis(DEFAULT_QUIET_MS),
        std::time::Duration::from_millis(DEFAULT_TIMEOUT_MS),
    );
    let snapshot = h.snapshot();
    assert_debug_snapshot!("home_screen", snapshot);

    h.send("q")?;
    h.expect_eof()?;
    Ok(())
}

#[test]
fn instant_screen_snapshot() -> Result<()> {
    let mut h = TuiHarness::spawn(DEFAULT_WIDTH, DEFAULT_HEIGHT)?;
    h.wait_for("\\[q\\] Quit")?;
    h.send("1")?;
    h.wait_for("Welcome to Glues :D")?;
    h.drain_until_quiet(
        std::time::Duration::from_millis(DEFAULT_QUIET_MS),
        std::time::Duration::from_millis(DEFAULT_TIMEOUT_MS),
    );
    let snapshot = h.snapshot();
    assert_debug_snapshot!("instant_screen", snapshot);

    h.send_ctrl_c()?; // Ctrl+C
    h.expect_eof()?;
    Ok(())
}
