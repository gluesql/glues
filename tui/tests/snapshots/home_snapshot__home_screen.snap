---
source: tui/tests/home_snapshot.rs
expression: screen
snapshot_kind: text
---
[?1049h[1;1H[38;5;240;48;5;253m Glues - TUI note-taking app offering complete data control an[38;5;77;48;5;253mî‚¾[38;5;0;48;5;77m [?] Show keymap
